### PR TITLE
improvement: update to php 7.1 and doctrine cache types

### PR DESCRIPTION
**Pull request by bycosta**
And also:
- Adds Travis CI
- Updates .gitignore
- Updates .php_cs
- Some other small improvements

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+install:
+  - travis_retry composer install --no-interaction --prefer-source
+
+script:
+  - vendor/bin/php-cs-fixer fix --dry-run -v
+  - vendor/bin/phpunit --verbose
+
+matrix:
+  fast_finish: true
```
Comment (nesl247) [inline .travis.yml]: Please fix the end of line issue.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+cache:
+  directories:
+    - vendor
+
+install:
+  - travis_retry composer install --no-interaction --prefer-source
+
+script:
+  - vendor/bin/php-cs-fixer fix --dry-run -v
```
Comment (nesl247) [inline .travis.yml]: Why not run these through composer?

Comment (bycosta) [reply .travis.yml]: No specific reason just copied this from another more up to date public lib of LinioIT.
There are also some differences.

Comment (nesl247) [reply .travis.yml]: I would use `composer lint:check` here. This way the same thing is used in CI as is used locally. It allows us to control it from a single place.

**Review by nesl247 — APPROVED** [review]

**Review by klaussilveira — APPROVED** [review]

**Pull request merged**